File: AQUECIMENTO_INTERI_2024/M/tudo.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, M; 
    cin >> N >> M;
    int L; 
    cin >> L;

    vector<vector<int>> fluxos(N, vector<int>(L, 0));


    // Lê os valores e separa nos respectivos vetores de fluxo
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < L; j++) {
            int valor; 
            cin >> valor;
            fluxos[i][j] = valor;
        }
    }

    int quadroAtual = 1;
    
    for (int posInicial = 0; posInicial < L; posInicial += M) {
        long long int somaQuadro = 0;

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                int pos = posInicial + j;
                if (pos < L) {
                    cout << fluxos[i][pos] << endl;
                    somaQuadro += fluxos[i][pos];
                }
            }
        }

        long long int mod = somaQuadro % 256;
        cout << mod << endl;

        quadroAtual++;
    }

    return 0;
}

File: AQUECIMENTO_INTERI_2024/M/tudo.py
import sys
input = sys.stdin.read

def main():
    data = input().strip().split()
    index = 0

    N = int(data[index])
    index += 1
    M = int(data[index])
    index += 1
    L = int(data[index])
    index += 1

    fluxos = []
    
    # Lê os valores e separa nos respectivos vetores de fluxo
    for i in range(N):
        fluxo = []
        for j in range(L):
            valor = int(data[index])
            index += 1
            fluxo.append(valor)
        fluxos.append(fluxo)

    quadroAtual = 1
    
    for posInicial in range(0, L, M):
        somaQuadro = 0

        for i in range(N):
            for j in range(M):
                pos = posInicial + j
                if pos < L:
                    print(fluxos[i][pos])
                    somaQuadro += fluxos[i][pos]

        mod = somaQuadro % 256
        print(mod)

        quadroAtual += 1

if __name__ == "__main__":
    main()
